ls1intum/Artemis
# `Programming exercises`: Show a warning when instructors add more than 15 tasks in a problem statement

### Checklist
#### General



#### Client
#### Changes affecting Programming Exercises

### Motivation and Context
This PR is related to and closes the issue [1070](https://github.com/ls1intum/Artemis/issues/1070).

### Description
The number of tasks should better not exceed 15 in a programming exercise. This PR:
- Doesn't complain if the number of tasks is less than or equal to 15 => ok message
- Adds a warning if the number of tasks exceeds 15 => warning message
- Correctly renders the step wizard of tasks regardless of their count

### Steps for Testing
Prerequisites:
- 1 Instructor or 1 Editor
- 1 Programming Exercise

1. Log in to Artemis
2. Navigate to Course Administration -> Exercises -> Click on a programming exercise -> Edit
3. Edit the problem statement and test for all the cases for the number of tasks less than/equal/greater than 15
4. Observe the ok/warning messages 

### Review Progress

#### Performance Review

#### Code Review
#### Manual Tests

### Screenshots

#### Tasks ok [0, 15]
<img width="1379" alt="image" src="https://github.com/ls1intum/Artemis/assets/81355607/15e663ae-a9b4-41b4-9af3-8baab6723eac">

#### Tasks warning [16, 50]
<img width="1385" alt="image" src="https://github.com/ls1intum/Artemis/assets/81355607/7d986cd3-c807-4550-b4fb-8004f6bd8bcb">

#### Step wizard with a large number of tasks
<img width="926" alt="image" src="https://github.com/ls1intum/Artemis/assets/81355607/e157daca-f1f1-4b58-b7dc-ef0ca029dd52">
<img width="924" alt="image" src="https://github.com/ls1intum/Artemis/assets/81355607/c8ef562f-bd64-4194-8b7a-12e34d55adfe">
